beetsplug/alternatives.py

```diff
@@ -16,18 +16,49 @@
 import argparse
 from concurrent import futures
 import six
+import traceback
 
 import beets
 from beets import util, art
 from beets.plugins import BeetsPlugin
 from beets.ui import Subcommand, get_path_formats, input_yn, UserError, \
     print_, decargs
 from beets.library import parse_query_string, Item
-from beets.util import syspath, displayable_path, cpu_count, bytestring_path
+from beets.util import syspath, displayable_path, cpu_count, bytestring_path, \
+        FilesystemError
 
 from beetsplug import convert
 
 
+def _remove(path, soft=True):
+    """Remove the file. If `soft`, then no error will be raised if the
+    file does not exist.
+    In contrast to beets' util.remove, this uses lexists such that it can
+    actually remove symlink links.
+    """
+    path = syspath(path)
+    if soft and not os.path.lexists(path):
+        return
+    try:
+        os.remove(path)
+    except (OSError, IOError) as exc:
+        raise FilesystemError(exc, 'delete', (path,), traceback.format_exc())
+
+
+def _samelink(p1, p2):
+    if p1 == p2:
+        return True
```

Comment:
What are the circumstances under which `p1 != p2` but they have the same `lstat`? Isn’t it enough to fully resolve `p1` and `p2` and assert that the are equal?

Reply:
Not sure, need to reconsider. I just copied over `_samefile` from `beets.util` and swapped `stat` for `lstat`.

Reply:
Ok. I think this logic is present in `beets.util.samefile` because we want to check that the target of the link is the same. In our case it should be enough to check that the resolved (i.e. absolute) paths are the same. No filesystem access required.

Reply:
Ok, so I should be able to replace the `_samelink(path, dest)` invocation with just `path == dest`, right? If I'm not missing anything, all these paths ultimately come from `Item.destination`, which returns absolute paths normalized through `os.path.normpath`.